src/http/api/follow.controller.ts

```diff
@@ -185,20 +201,33 @@ export class FollowController {
             });
         }
 
-        // Need to get the follow
+        await this.accountService.unfollowAccount(
+            unfollowerAccount,
+            accountToUnfollow,
+        );
+
+        // If the account to unfollow is internal, we can just unfollow it directly
+        // without federating the unfollow
+        if (accountToUnfollow.isInternal) {
+            return new Response(null, {
+                status: 202,
+            });
+        }
```

Comment:
question: should we be returning a different response body, based on whether we send an activity or not?

We’re now returning:
- empty body, if we don’t send the Unfollow activity
- `unfollow.toJsonLd()` if we do

I wonder if we should stick to returning the same response body in both cases, so that whether we send an activity or not is an internal implementation detail, and not something the client needs to care about?

(We do that in the new follow logic, the branching is on whether to send the activity or not, but we return the same response)

Reply:
Yeh i should of made a note on this - I had a look throughout the codebases and we are not actually using the response from this endpoint so there is no need to return anything, but agree the branching logic creates inconsistency.  I've updated this to now just return an empty body in all cases → https://github.com/TryGhost/ActivityPub/pull/1399/commits/1470201c36271b0ff7d3672da3d720702642baaa

The client should continue to work fine without any changes as it should handle the 204 gracefully → https://github.com/TryGhost/Ghost/blob/main/apps/activitypub/src/api/activitypub.ts#L248-L250

Side note: we wasn't even typing the response correctly for the unfollow endpoint in the client 😅 https://github.com/TryGhost/Ghost/blob/main/apps/activitypub/src/api/activitypub.ts#L321-L325

Reply:
💯 Nice one!